config/component_macros.cmake

```diff
@@ -43,6 +43,24 @@ set(Draco_std_target_props_CUDA
 set(Draco_std_target_props
   INTERPROCEDURAL_OPTIMIZATION_RELEASE ${USE_IPO}
   POSITION_INDEPENDENT_CODE ON )
+# Throttle use of GPU. We don't want too many test sharing the same GPU.
+if( NOT DEFINED MAX_TESTS_PER_GPU )
+  set( MAX_TESTS_PER_GPU 10 CACHE STRING "max concurrent gpu tests" FORCE)
```

Comment:
Tests that need a GPU will be assigned a resource lock named `GPU_<N>` where `N < MAX_TESTS_PER_GPU` and is incremented after each test is registered. In this case, up to 10 concurrent tests could be using the GPU.